factory_reset/factory_reset.proto

```diff
@@ -47,6 +62,8 @@ message ResetError {
   bool factory_os_unsupported = 1;
   // Zero fill is not supported.
   bool zero_fill_unsupported = 2;
+  // Management config is not supported.
+  bool mgmt_config_unsupported = 5;
```

Comment:
I know this has been flagged previously by Marcus, but I'll mention it again here in the interest of having the discussion more broadly.  This field won't be a reliable indicator of what's going to happen.

Old devices that aren't aware of the `mgmt_config` field in the request also won't be aware of the `mgmt_config_unsupported` field in the response and therefore won't be able to set it to indicate their lack of awareness.  Instead, they'll just silently ignore `mgmt_config` altogether.  You won't be able to distinguish between an old device that ignored this part of your request and a new device that processed it like you requested.

Splitting the boot parameters from the `FactoryReset` service (like was proposed in another comment on the pull request) would make it easier to know what's going on.  If you sent an old device something like `SetBootstrapServerList` in a `BootParameters` service, it would respond with an unimplemented error.

Reply:
Bump the version of the service. Clients then have the option to ensure the service implements the new message.

Reply:
>Bump the version of the service. Clients then have the option to ensure the service implements the new message.

gnoi package names don't include versions, hence it will not be visible

Reply:
I am not sure if we are talking about the same thing, but you should be able to query service options via file descriptors via reflection if the server implementes it. Its convoluted, but its there.

https://github.com/openconfig/gnoi/blob/bae5fb4f4d3742873b361f2cb3867e4d5ef1e06c/types/types.proto#L26

Reply:
I see what you mean. Indeed, I was thinking about a different thing (traditional grpc versioning, where the version is part of the namespace/package name). Thx for clarifying.